crates/iceberg/src/transform/temporal.rs

```diff
@@ -175,7 +200,7 @@ impl TransformFunction for Day {
                 .as_any()
                 .downcast_ref::<TimestampMicrosecondArray>()
                 .unwrap()
-                .unary(|v| -> i32 { Self::day_timestamp_micro(v) }),
+                .unary(|v| -> i32 { Self::day_timestamp_micro(v).unwrap() }),
```

Comment:
@liurenjie1024 
can we handle the unwrap() properly, haven't found a way yet. But perhaps you have an idea?